src/test/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriterRoundtripTest.java

```diff
@@ -56,124 +62,160 @@ public static Stream<File> data() throws Exception {
+                                sourceField.tag == ExifTagConstants.EXIF_TAG_INTEROP_OFFSET.tag ||
+                                sourceField.tag == TiffTagConstants.TIFF_TAG_JPEG_INTERCHANGE_FORMAT.tag;
+
+                /*
+                 * Ignore offset fields. They may not be needed after rewrite
+                 * and their value changes anyway.
+                 */
+                if (isOffsetField)
+                    continue;
+
+                TiffOutputField duplicateField = duplicateDirectory.findField(sourceField.tag);
+
+                assertNotNull(duplicateField, "Field is missing: " + sourceField.tagInfo);
+
+                assertEquals(sourceField.tag, duplicateField.tag, "TiffOutputField tag mismatch.");
+                assertEquals(sourceField.abstractFieldType, duplicateField.abstractFieldType, "TiffOutputField fieldType mismatch.");
+                assertEquals(sourceField.count, duplicateField.count, "TiffOutputField count mismatch.");
+
+                boolean sameValue = TiffOutputFieldTestUtil.bytesEqual(sourceField, duplicateField);
+
+                assertTrue(sameValue, "Bytes are different for field: " + sourceField);
             }
         }
     }
 
     private void copyToDuplicateFile(final File sourceFile, final TiffOutputSet duplicateTiffOutputSet) throws IOException, ImagingException, ImagingException {
+
         final ExifRewriter exifRewriter = new ExifRewriter();
+
         duplicateFile = createTempFile();
+
         try (OutputStream duplicateOutputStream = new BufferedOutputStream(new FileOutputStream(duplicateFile))) {
             exifRewriter.updateExifMetadataLossless(sourceFile, duplicateOutputStream, duplicateTiffOutputSet);
         }
     }
 
     private File createTempFile() {
+
         final String tempDir = FileUtils.getTempDirectoryPath();
         final String tempFileName = this.getClass().getName() + "-" + random.nextLong() + ".tmp";
+
         return new File(tempDir, tempFileName);
     }
 
     private TiffOutputSet duplicateTiffOutputSet(final TiffOutputSet sourceTiffOutputSet) throws ImagingException {
-        final TiffOutputSet duplicateTiffOutputSet = new TiffOutputSet();
+
+        final TiffOutputSet duplicateTiffOutputSet = new TiffOutputSet(
+                sourceTiffOutputSet.byteOrder
+        );
+
         for (final TiffOutputDirectory tiffOutputDirectory : sourceTiffOutputSet) {
             duplicateTiffOutputSet.addDirectory(tiffOutputDirectory);
         }
+
         return duplicateTiffOutputSet;
     }
 
-    private JpegImageMetadata getJpegImageMetadata(final File sourceFile) throws ImagingException, IOException {
+    private JpegImageMetadata getJpegImageMetadata(final File sourceFile) throws IOException {
+
         final JpegImageMetadata jpegImageMetadata = (JpegImageMetadata) Imaging.getMetadata(sourceFile);
-        if (null == jpegImageMetadata) {
+
+        if (jpegImageMetadata == null) {
```

Comment:
Don't clutter a PR changes in style, please, it clutters the PR with useless changes and takes longer to review.

Reply:
It actually improves the code style. I see you doing a lot of code style changes in the commit history. So you should welcome a change like that.